lib/auth/auth_with_roles.go

```diff
@@ -2452,6 +2452,8 @@ func (a *ServerWithRoles) ListAccessRequests(ctx context.Context, req *proto.Lis
 	if req.Filter == nil {
 		req.Filter = &types.AccessRequestFilter{}
 	}
+	// set the requesting user to be used in the filter match
+	req.Filter.Requester = a.context.User.GetName()
```

Comment:
the request for a user is with `req.Filter.User`, requester is "api call requester" which is the context of the user making the call. it may not be the best name i suppose, but since all the matching happens on the filter, this is where it had to go.

Your suggestion is fine, but it wouldn't overwrite anything so doesn't matter where this definition goes imo

Reply:
gotcha, i'd just include in the comment that it is meant to be overwritten